.nginx/nginx.conf

```diff
@@ -0,0 +1,15 @@
+worker_processes 4;
+
+events { worker_connections 1024; }
+
+http {
+    server {
+        listen 3000;
+        root  /usr/share/nginx/html;
+        include /etc/nginx/mime.types;
+
+        location /appui {
```

Comment:
What exactly is /appui? I don't particularly understand why this particular word is used. I'm assuming it's a _keyword?_

Reply:
We could use any word here. Doesn't have to be `/appui`. Here's the flow of command that takes place when you navigate to the website as deployed right now. User --> Load Balancer --> Kubernetes (skip if deployed on cloud run) --> container --> nginx --> React router. In this situation you see that in order for the url to get all the way to the react application one of the things it needs to pass through is nginx. What's going on here is that we're saying take whatever url you are given and go straight to `index.html` no matter what. This ensures that the React Router will handle everything. 

In many web applications people will specify saying if a user goes to `/site` then serve up some html, if they go to `/file` then serve up some content or a different application and maybe they have a wildcard catch all at the end to serve a 404 page. You can basically think of the location blocks as performing the same task as our react router but abstracted one step up. We don't really need it to do anything so we're just saying try the path at `index.html` and let our application handle what to display.

Check out this resource for more understanding on how server and location blocks work in nginx [here](https://www.digitalocean.com/community/tutorials/understanding-nginx-server-and-location-block-selection-algorithms)